session.go

```diff
@@ -434,7 +445,6 @@ func (s *Session) mux(remoteBegin *performBegin) {
 				fr.OutgoingWindow = s.outgoingWindow
 				debug(1, "TX(Session) - tx: %s", fr)
 				s.txFrame(fr, nil)
-				remoteOutgoingWindow = s.incomingWindow
```

Comment:
as noted in #15 this was not as per protocol. the remoteOutgoingWindow is supposed to reflect the remote peer state. we cannot just updated it to incomingWindow.
the remotePeer lets us know where it's at in flow frames, and that's the only place we should update it.